### ISSUE TITLE
Use Graphql to reduce server requests?

### ISSUE THREAD

**Issue by tyler-barnes** [postlight/headless-wp-starter#43]
Graphql is a rest api replacement which lets you define which data you want to retrieve. Instead of this starter making a separate request for everything on the page, a single request could be made and then the data could be passed into all the places that are currently making individual requests. I think this should improve performance by a lot but I'm not a graphql expert. There is a wordpress plugin for graphql https://www.wpgraphql.com/

**Comment by ginatrapani** [comment]
Yeah, I started playing with this in this branch:  https://github.com/postlight/headless-wp-starter/tree/graphql-test

Currently there's a bug in wp-graphql that makes it incompatible with `wp server` (or any proxying). I'm talking to the wp-graphql folks now about it. For reference, the issue: https://github.com/wp-graphql/wp-graphql/issues/355

**Comment by tyler-barnes** [comment]
Oh, that's exciting! This starter is going to be the best

**Comment by ginatrapani** [comment]
That bug has been resolved. I'm doing more work with wp-graphql lately, and while it still has a way to go before its 1.0 release, I think it's worth including here, and updating the README -- and maybe the frontend? -- for sure.

**Comment by tyler-barnes** [comment]
@ginatrapani that's awesome! I find this starter slow to respond sometimes and I assume it's due to the number of requests that happen before next.js sends anything to the browser. I think it's definitely worth including and adding to the frontend as having a single request to the WP install should speed things up.

**Comment by meishern** [comment]
An issue related to this, is that when viewing source, right now will show: `__NEXT_DATA__ = {"props"`
Which contains all sorts of private data which should not be displayed which includes (1) the URL of the body section of the headless-wp, which should never be known to the end user, as well as; (2) the login name of the wordpress user and the corresponding id in the wordpress database, which is a security hole. 

By using GraphQL, this private data should no longer be shown.

**Comment by TylerBarnes** [comment]
Any progress or roadblocks on this?

**Comment by ginatrapani** [comment]
> Any progress or roadblocks on this?

No, I've just been caught up with other work (a lot of it was directly with wp-graphql!) but I'm back to this. I may have been overthinking it, so maybe I'll just commit a version that serves GraphQL from the backend, and work on the frontend example code separately. More soon!

**Comment by TylerBarnes** [comment]
So cool! I can't wait to find some more time to try this out!
